### ISSUE TITLE
Some threads stuck while others are not

### ISSUE THREAD

**Issue by JanStreffing** [UNICORE-EU/uftp#15]
While I was very optimistic about the transfer speeds between aleph (ICCP, ROK) and levante (DKRZ, DE) after #14,
I'm now faced with a new issue. It seems that with the new settings at DKRZ some threads get stuck. 
using:
```
uftp cp --identity $UFTP_KEY --streams 4 -t20 -T 1G -v atm_reduced_3h_pl_vo_3h_pl_195106-195106.nc $UFTP_AUTH_URL//work/ab0246/a270092/downloads/TCO319-DART/1950-CTRL/
```

I obtain:
![image](https://github.com/UNICORE-EU/uftp/assets/25483665/166eaec5-c9ae-4857-a59d-ef59fdd1a90a)

Not that threads number 5, 7,8 and 10 have a much lower transfer speed. The transferspeeds of the other 16 threads were fluctuating for about 1 minute, as they were active until that part of the file was send. The four threads in question froze at their displayed speeds after just a second or so. Now the transfer has been stuck at 75% for more than 10 minutes. I suspect these four threads somehow got hung up just after initialization.

Any idea what can cause this behaviour or how to investigate further? @BerndSchuller, @ocarino

**Comment by BerndSchuller** [comment]
hi,

TCP connections can kind of stall once packets start getting lost. I'd recommend using less streams and less connections.

**Comment by BerndSchuller** [comment]
... keep open to check whether the error handling / error reporting in v1.4.4 and in general is OK, just to be safe ;)
